jobrunner/lib/docker.py

```diff
@@ -184,16 +184,22 @@ def copy_from_volume(volume_name, source, dest, timeout=None):
     not, by default, have any timeout.
     """
     dest.parent.mkdir(parents=True, exist_ok=True)
-    docker(
-        [
-            "cp",
-            f"{manager_name(volume_name)}:{VOLUME_MOUNT_POINT}/{source}",
-            dest,
-        ],
-        check=True,
-        capture_output=True,
-        timeout=timeout,
-    )
+    tmp = dest.with_suffix(dest.suffix + ".tmp")
```

Comment:
Given that we're doing this in `lib.docker` (which makes sense as it means we get the behaviour for free everywhere else) I've got a very tiny worry about the fixed `.tmp` suffix. We don't know in what contexts we'll end up using this function and it's possible we'll end up blatting something we don't want to.

What about something super crude like:
```py
tmp = dest.with_suffix(f"{dest.suffix}.{secrets.token_hex(8)}.tmp")
```

Reply:
good call, fixed